### PR TITLE
CDPAM-3145: DataLake Creation Failed

### PR DESCRIPTION
**Pull request by zabhi**
To reach out to https endpoints, golang depends on the value in `https_proxy` variable. Earlier, golang used to fall back to `http_proxy` variable inn the absence of an `https_proxy` variable.
After a recent golang version update (1.16), the behavior was changed to stop the client falling back to `http_proxy` variable. This broke our NTP proxy support.

This change fixes the problem by setting the correct variable.

### CODE REVIEW

File: saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh [hunk 1]
```diff
@@ -182,7 +182,7 @@ setup_proxy() {
     fi
     PROXY_ENV_FILE=/etc/cdp/proxy.env
     mkdir -p /etc/cdp
-    echo ${PROXY_PROTOCOL}_proxy=$PROXY_URL > $PROXY_ENV_FILE
+    echo https_proxy=$PROXY_URL > $PROXY_ENV_FILE
```
Comment (SachinHg) [inline saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh:185]: what about if the proxy is of type `http_proxy`?

Comment (lacikaaa) [reply saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh:185]: yes this seems odd, I would expect to use the correct protocol. maybe we should set both? is there any automated tests for these?

Comment (zabhi) [reply saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh:185]: The variable name is tied to the target URL and not the proxy' protocol. So, the original code was incorrect.

Earlier, golang used to fall back to `http_proxy` variable in the absence of an `https_proxy` variable.
After a recent golang version update (1.16), the behavior was changed to stop the client falling back to `http_proxy` variable. This broke our NTP proxy support. (See https://go.dev/doc/go1.16#library)

So, if the client needs to connect to `https://example.com`, the variable used would be `https_proxy`, even if the proxy address uses http (eg. `http://proxy.host:5138`)

**Review by lacikaaa — CHANGES_REQUESTED** [review]
please squash your commits

**Review by lacikaaa — APPROVED** [review]

**Review by SachinHg — APPROVED** [review]
lgtm

**Comment by meetrajsinhvala** [comment]
We have baked freeipa images from this branch and below are the image-UUIDs for different cloud providers.

- AWS : 90fa3754-121d-4b08-88cd-39d7cc4080e2
- Azure : bb13bbe6-8afc-454f-b370-aecb05cbb9fa 
- GCP : caf361ec-1089-4a4c-ab41-ca6e77dd95c6

We have tested this with our CCMv2 dev test suite and CCMv2 e2e test suite (with proxies). CCMv2 dev test runs and e2e tests run were successful.

**Pull request merged**